### PR TITLE
PL-1089: implementing changes to getBalance/getTransaction

### PR DESCRIPTION
**Pull request by VKCR**

### CODE REVIEW

File: src/http/AuthContext.js [hunk 1]
```diff
@@ -26,6 +38,17 @@ class AuthContext {
     get onBehalfOf() {
         return this._onBehalfOf;
     }
+
+    /**
+     * Gets the key level and resets it to low
+     *
+     * @return {string} keyLevel - key level being used
+     */
+    get keyLevel() {
```
Comment (lahosken) [inline src/http/AuthContext.js]: a property-getter that changes the value of that property seems likely to confuse us later.

what do you think of instead having methods

// Sets key level to use signing the next auth header (will revert back to LOW afterwards)
setNextKeyLevel(keyLevel)

// Get key level to use signing the next auth header (and revert value back to LOW)
getNextKeyLevel()

Comment (mariano54) [reply src/http/AuthContext.js]: I agree with larry. Getter should not modify the object.

File: src/sample/MemberGetBalanceSample.js [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
  */
+
 export default async (member) => {
     var sums = {};
     const accounts = await member.getAccounts();
     for (var i = 0; i < accounts.length; i++) {
-        const balance = await member.getBalance(accounts[i].id);
-        const currency = balance.available.currency;
+        const balance = await member.getBalance(accounts[i].id, config.KeyLevel.STANDARD);
+        const currency = balance.balance.available.currency;
```
Comment (mariano54) [inline src/sample/MemberGetBalanceSample.js]: We shouldn't have two balances there.

File: src/main/Member.js [hunk 3]
```diff
@@ -798,11 +800,12 @@ export default class Member {
      * @param {string} accountId - id of the account
      * @param {string} offset - where to start looking
      * @param {int} limit - how many to retrieve
+     * @param {string} keyLevel - key level
      * @return {Promise} transactions - Transactions
      */
-    getTransactions(accountId, offset, limit) {
+    getTransactions(accountId, offset, limit, keyLevel) {
```
Comment (mariano54) [inline src/main/Member.js]: can we make keyLevel optional? If it's equal to undefined, set to LOW or something.

File: src/http/AuthContext.js (hunk 1 above)
Comment (VKCR) [reply src/http/AuthContext.js]: moved the logic to AuthHeader._keyLevel(context)

File: src/sample/MemberGetBalanceSample.js (hunk 2 above)
Comment (VKCR) [reply src/sample/MemberGetBalanceSample.js]: Renamed balance => balanceResponse

File: src/main/Member.js (hunk 3 above)
Comment (VKCR) [reply src/main/Member.js]: It's already the case, in keyLevel's setter it sets it to LOW if it's undefined.

File: src/http/AuthHeader.js [hunk 4]
```diff
@@ -85,6 +85,15 @@ class AuthHeader {
         }
         return '';
     }
+
+    static _keyLevel(context) {
```
Comment (lahosken) [inline src/http/AuthHeader.js]: += comment so we can remember what's going on later
// if we specified that this request should use a keyLevel > LOW, we notice that here
// (and re-set the keyLevel for the next request)

Comment (VKCR) [reply src/http/AuthHeader.js]: added the comment

**Pull request merged**